src/hashes.c

```diff
@@ -24,6 +24,10 @@ static const struct hash_info {
 	{ MH_H_BLAKE2S, "blake2s", 32 }
 };
 
+// REVIEW NOTE:
+// This could just be a table where the index into the array is the hash
+// function.  This would make the lookups much simpler.
+
```

Comment:
I think having NULL's should be fine. We can place a bunch of them on the same line to make it orderly and use comments to denote indexes to make it more readable